pkg/kds/v2/store/sync.go

```diff
@@ -278,6 +285,14 @@ func newIndexed(rs core_model.ResourceList) *indexed {
 	return &indexed{indexByResourceKey: idxByRk}
 }
 
+func indexOfResourceKeys(rks []core_model.ResourceKey) map[core_model.ResourceKey]struct{} {
```

Comment:
I'm surprised such a function doesn't exist already.

Reply:
https://github.com/kumahq/kuma/blob/d7ec0a2b1ac19208fb7dd9726309e3cf8cdc5848/pkg/core/resources/model/resource.go#L764-L771 is similar no?

Reply:
If there's no possibility to adapt this helper should probably move next to the method above